src/audio/mux/mux.c

```diff
@@ -200,7 +205,229 @@ static struct comp_dev *mux_new(const struct comp_driver *drv,
 	dev->state = COMP_STATE_READY;
 	return dev;
 }
+#else
+static int build_config(struct comp_data *cd, struct comp_dev *dev)
+{
+	dev->ipc_config.type = SOF_COMP_MUX;
+	cd->config.num_streams = 2;
+	int mask = 1;
+	int i, j;
+
+	/* clear masks */
+	for (i = 0; i < cd->config.num_streams; i++)
+		for (j = 0 ; j < PLATFORM_MAX_CHANNELS; j++)
+			cd->config.streams[i].mask[j] = 0;
+	/*
+	 * Setting masks for streams. Channels for IPC4 are mapped
```

Comment:
Any documentation of the IPC messages should be maintained in the headers and written in doxygen (javadoc) format. The IPC interface specification will be generated from the headers and included in the sof-docs.